### PR TITLE
Move published files list out of published branch API into published files API

### PR DESCRIPTION
**Pull request by jmhooper**
This commit adds a new route which is used to list the files published to a branch. This new route corresponds to a "published files" API and is ~`/v0/site/{site_id}/published-branch/{branch}/file`~ `/v0/site/{site_id}/published-branch/{branch}/published-file`

This renders a list of objects that correspond to the files published to S3 for the given branch.

This commit also removes this information from the published branch `findOne` action.

### CODE REVIEW

**Comment by jmhooper** [comment]
This PR is WIP. The backend works is done and needs to be tied into the frontend.

In addition, I'm considering changing the route to `/v0/site/1/published-branch/master/published-file`.

**Comment by jmhooper** [comment]
The path has been updated and the corresponding changes have been made to the frontend. This PR is no longer WIP and it review ready.

File: public/swagger/index.yml [hunk 1]
```diff
@@ -356,7 +356,7 @@ paths:
       type: string
       required: true
     get:
-      summary: Fetch information about what's pusblished for the site at the given branch
+      summary: Fetch metadata about what's pusblished for the site at the given branch
```
Comment (brandocalrissian) [inline public/swagger/index.yml]: pusblished --> published

File: test/api/requests/published-branch.test.js [hunk 2]
```diff
@@ -96,7 +95,7 @@ describe("Published Files API", () => {
       }).catch(done)
     })
 
-    it("should list the files on S3 for a given branch", done => {
+    it("should render a JSON response for a pubslished branch", done => {
```
Comment (brandocalrissian) [inline test/api/requests/published-branch.test.js]: pubslished -> published

**Pull request merged**